### PR TITLE
Add a tag tab + static tag list

### PR DESCRIPTION
**Pull request by AlexAndBear**
## Description

1. Introduce a new tab, where the tag input field will be displayed.
1.1 Keyboard navigation in the input field's dropdown works now proper, in the past the user needed to hover over on item in the dropdown before the arrow keys could be used

2. Replace the tag input field in the file list by a read only tag list.
2.1 Improving tag appearance to clarify that these are tags and no shares.
2.2 Clicking on a tag opens the tag tab.
2.3 Tag tab and tag list are in sync, means that editing (select, unselect, rename, remove) tags will appear in the tag list immediately.

## Related Issue

- Fixes https://github.com/owncloud/enterprise/issues/4294

## Screenshots (if appropriate):
Screenshot_20201211_123331

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [x] Documentation ticket raised: https://github.com/owncloud/docs/issues/2917 
- [x] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by phil-davis** [comment]
The drone agents + docker "have a bad day" sometimes. https://drone.owncloud.com/owncloud/core/27900/152/8
```
Error response from daemon: No such image: owncloudci/php:7.4
```

That is rubbish - the docker image should be pulled from docker if the agent does not already have it.

I restarted drone CI.

**Comment by JanAckermann** [comment]
> The drone agents + docker "have a bad day" sometimes. https://drone.owncloud.com/owncloud/core/27900/152/8
> 
> ```
> Error response from daemon: No such image: owncloudci/php:7.4
> ```
> 
> That is rubbish - the docker image should be pulled from docker if the agent does not already have it.
> 
> I restarted drone CI.

I think today drone is not willing to work properly

```
+ php occ security:certificates:import /drone/server.crt
--
188 | certificate not found
```

**Comment by JanAckermann** [comment]
TODO: Fix acceptance tests

File: core/js/systemtags/systemtagsinputfield.js [hunk 1]
```diff
@@ -259,6 +261,17 @@
 		 * @param {Object} query select2 query object
 		 */
 		_queryTagsAutocomplete: function(query) {
+			var termErrors = false;
+
+			if(query.term.indexOf('@') !== -1) {
+				termErrors = true;
+				this.$errorField.text(t('systemtags', 'This looks like an email address or federated cloud ID. Are you sure you want to use a tag?')).removeClass('hidden');
```
Comment (mmattel) [inline core/js/systemtags/systemtagsinputfield.js:268]:
```suggestion
				this.$errorField.text(t('systemtags', 'This looks like an email address or federated cloud ID. Are you sure you want to use this tag name?')).removeClass('hidden');
```
You are not arguing about using a tag in general but about the particular name used

**Comment by mmattel** [comment]
Some test have not started, services not ready...
```
+ wait-for-it -t 600 fsweb.test.owncloud.com:445
--
59 | services aren't ready in 10m0s
```

**Comment by JanAckermann** [comment]
> Some test have not started, services not ready...
> 
> ```
> + wait-for-it -t 600 fsweb.test.owncloud.com:445
> --
> 59 | services aren't ready in 10m0s
> ```
Yes, drone ist really frustrating today....

**Comment by phil-davis** [comment]
`fsweb.test.owncloud.com` was down. The sysadmins have got it up again. I restarted drone CI.

**Comment by JanAckermann** [comment]
> `fsweb.test.owncloud.com` was down. The sysadmins have got it up again. I restarted drone CI.

Thank you, checks passed now

**Comment by mmattel** [comment]
@JanAckermann 
I have tried it and it works great.
The following came to my mind when trying an eMail address as tag:
- Yes, you get the info "This looks like an eMail...."
But what do I do with that info, which consequences arise?
Pressing enter adds the tag, what do I have to fear if I have done so?
- `This looks like an email address or federated cloud ID. Are you sure you want to use this tag name?`
needs improvement, because you are checking the `@` character. -->
`This looks like you start typing an email address or federated cloud ID. Are you sure you want to use this kind tag name?`
- When you start typing the tag with `@` like `@JanAckermann`, this does NOT look like an eMail address. Where I am not sure if that would be a valid start for a federated share. A federated share, as far I remember, starts with `username@instance-url`. Maybe you want to differenciate this in the text shown (text = case dependent). Saying that, I see that there are three cases. `@` (alone), `name@` and `@name` ...
- It would be great, if the editable tag list would open not only when clicking the mouse into the horizontal tag list, but also on the tag itself.

**Comment by JanAckermann** [comment]
@mmattel Thank you for your Input!

> Yes, you get the info "This looks like an eMail...."
But what do I do with that info, which consequences arise?

@pmaier1 @micbar I think we could really get rid of this message, from now it is so clear that these are tags, what do you think?

> It would be great, if the editable tag list would open not only when clicking the mouse into the horizontal tag list, but also on the tag itself.
I don't get it, could you explain maybe with pictures :) ?

**Comment by micbar** [comment]
@mmattel 

> Yes, you get the info "This looks like an eMail...."
> But what do I do with that info, which consequences arise?

The problem which started this PR is, that many users enter a username in the tags field and think that they have shared the file. The instance we are talking about has a lot of federated shares.

But i would suggest to rather remove the message at all. It was an idea to help the users, but seems to be not as helpful as we thought.

@JanAckermann @JammingBen agree?

**Comment by JammingBen** [comment]
> The problem which started this PR is, that many users enter a username in the tags field and think that they have shared the file. The instance we are talking about has a lot of federated shares.

> But i would suggest to rather remove the message at all. It was an idea to help the users, but seems to be not as helpful as we thought.

> @JanAckermann @JammingBen agree?

I totally agree. In the future, users have to click either on the "tags"-tab, or on the tags directly which will also open the "tags"-tab. So IMO this makes in pretty clear that we are dealing with tags here, not with email addresses.

**Comment by JanAckermann** [comment]
Message removed

**Comment by mmattel** [comment]
@JanAckermann 
It would be great, if it would be possible to edit a particular tag by just clicking on it versus clicking in a free part of the total (horizontal) list to open the full editable list of tags and then pick the one to change.
At least, clicking on any item in the horizontal list to open the editable list (compared to now where you need to click on the right free space of the horizonal list)

**Comment by JanAckermann** [comment]
> @JanAckermann
> It would be great, if it would be possible to edit a particular tag by just clicking on it versus clicking in a free part of the total (horizontal) list to open the full editable list of tags and then pick the one to change.
> At least, clicking on any item in the horizontal list to open the editable list (compared to now where you need to click on the right free space of the horizonal list)

Unfortunately Select2 3.5 stops the propagation of most events out of the container so I am not able to catch the events when a tag is just clicked, with Select2 v4, this is no issue any more but needs a bunch of refactoring. I think this would blow up the frame of this issue.

**Comment by mmattel** [comment]
Select2 v4 is out of reach of this PR, totally agree. You may file another PR for that...

And CI is making us crazy...
What the h**l is going on atm?
```
Error: connect ETIMEDOUT 52.216.243.76:443
--
28 | at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1191:14)
```

**Comment by JanAckermann** [comment]
> Select2 v4 is out of reach of this PR, totally agree. You may file another PR for that...
> 
> And CI is making us crazy...
> What the h**l is going on atm?
> 
> ```
> Error: connect ETIMEDOUT 52.216.243.76:443
> --
> 28 | at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1191:14)
> ```

Basically the the lisence of the windows server has been expired, so the server shuts down automatically. 
Cristian already takes care about this problem

**Comment by mmattel** [comment]
🤦‍♂️ 
When do you think it is up again?

**Comment by JanAckermann** [comment]
> When do you think it is up again?

I have no clue 🧐

File: changelog/unreleased/38197 [hunk 2]
```diff
@@ -0,0 +1,11 @@
+Enhancement: Improve file list tag management
+
+1. Introduce a new tab, where the tag input field will be displayed
+1.1 Keyboard navigation in the input field's dropdown works now proper, in the past the user needed to hover over on item in the dropdown before the arrow keys could be used
```
Comment (phil-davis) [inline changelog/unreleased/38197:4]:
```suggestion
1.1 Keyboard navigation in the input field's dropdown works correctly now. In the past the user needed to hover over an item in the dropdown before the arrow keys could be used
```

File: changelog/unreleased/38197 [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+2. Replace the tag input field in the file list by a read only tag list
+2.1 Improving tag appearance to clarify that these are tags and no shares
```
Comment (phil-davis) [inline changelog/unreleased/38197:7]:
```suggestion
2.1 Improving tag appearance to clarify that these are tags and not shares
```

File: changelog/unreleased/38197 [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+2.2 Clicking on a tag opens the tag tab
+2.3 Tag tab and tag list are in sync, means that editing (select, unselect, rename, remove) tags will appear in the tag list immediately
```
Comment (phil-davis) [inline changelog/unreleased/38197:9]:
```suggestion
2.3 Tag tab and tag list are in sync. Editing (select, unselect, rename, remove) tags will appear in the tag list immediately.
```

**Review by phil-davis — COMMENTED** [review]
Minor comments on the changelog

**Review by mmattel — APPROVED** [review]

**Pull request merged**